pilot/cmd/pilot-agent/status/server.go

```diff
@@ -707,10 +707,13 @@ func (s *Server) handleAppProbeHTTPGet(w http.ResponseWriter, req *http.Request,
 	}
 
 	// Forward incoming headers to the application.
-	appReq.Host = req.Host
+	if prober.HTTPGet.Host != "" {
```

Comment:
Can you add a release note for this one, @jaellio? I think it's better to tell users what has been fixed instead of silently merging.

Reply:
Yes, good call! Will add.

Reply:
@hanxiaop is it possible to add the release note to this PR? I created a new PR on master with the release note #45905

Reply:
I'm not sure how to add content to istio-testing's PR either. What I used to do was a manual cherry-pick, or you can cherry-pick the release note and let both be merged together.

Reply:
@jaellio your release note got cherry-picked in a separate PR #45927. I'll approve it once this merges